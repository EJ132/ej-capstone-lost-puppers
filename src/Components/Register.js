import React, { Component } from 'react'
import '../Stylesheets/Login.css'
import paw from '../Images/paw.png'
import NavBar from './Navbar'

export default class Register extends Component {

    handleRegister = event => {
        event.preventDefault()
        const { full_name, user_name, password, re_enter_password } = event.target
    
        full_name.value = ''
        user_name.value = ''
        password.value = ''
        re_enter_password.value = ''
        // this.props.onRegistrationSuccess()
      }
      

    render(){

        return (
            <div>

                <NavBar />

                <section className="loginBG">
                    <form className="login" onSubmit={this.handleRegister}>
                        <img src={paw} alt="paw"/>
                        <label>Name</label>
                        <input placeholder="EJ Gonzalez" name="full_name" type="fullname" required></input>
                        <label>Username</label>
                        <input placeholder="puppy_lover132" name="user_name" type="user_name" required></input>
                        <label>Password</label>
                        <input placeholder="********" name="password" type="password" required></input>
                        <label>Re-enter Password</label>
                        <input placeholder="********" name="re_enter_password" type="password" required></input>
                        <button type="submit">Submit</button>
                    </form>
                    <p className="validation">{this.handleRegister}</p>
                </section>

                <footer>&#169; EJ Gonzalez 2019</footer>
            </div>
        )
    }
}