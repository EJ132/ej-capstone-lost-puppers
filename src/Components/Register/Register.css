.validation {
    color: black;
    position: absolute;
    bottom: 100px;
    right: 0;
}