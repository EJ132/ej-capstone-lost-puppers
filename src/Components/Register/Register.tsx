import * as React from "react";
import "../Login/Login.css";
import paw from "../../Images/paw.png";
import NavBar from "../NavBar/Navbar";
import AuthApiService from "../../services/auth-api-service";
import history from "../../Context/history";

export default class Register extends React.Component {
  state = {
    error: null,
  };

  handleRegister = (event) => {
    event.preventDefault();
    const { fullname, user_name, password, re_enter_password } = event.target;

    this.setState({ error: null });

    // eslint-disable-next-line eqeqeq
    if (password.value !== re_enter_password.value) {
      return this.setState({
        error: "Please enter matching passwords",
      });
    }

    AuthApiService.postUser({
      user_name: user_name.value,
      password: password.value,
      fullname: fullname.value,
    })
      .then((user) => {
        fullname.value = "";
        user_name.value = "";
        password.value = "";
        re_enter_password.value = "";
        // this.props.onRegistrationSuccess()
      })
      .then(() => history.push("/login"))
      .catch((res) => {
        this.setState({ error: res.error });
      });
  };

  render() {
    const { error } = this.state;

    return (
      <div>
        <NavBar />

        <section className="loginBG">
          <form className="login" onSubmit={this.handleRegister}>
            <section className="regError" role="alert">
              {error && <p className="red">{error}</p>}
            </section>
            <img src={paw} alt="paw" />
            <label>Name</label>
            <input
              placeholder="EJ Gonzalez"
              name="fullname"
              type="text"
              required
            ></input>
            <label>Username</label>
            <input
              placeholder="puppy_lover132"
              name="user_name"
              type="user_name"
              required
            ></input>
            <label>Password</label>
            <input
              placeholder="********"
              name="password"
              type="password"
              required
            ></input>
            <label>Re-enter Password</label>
            <input
              placeholder="********"
              name="re_enter_password"
              type="password"
              required
            ></input>
            <button type="submit">Submit</button>
          </form>
          {/* <p className="validation">{this.handleRegister}</p> */}
        </section>

        <footer>&#169; EJ Gonzalez 2019</footer>
      </div>
    );
  }
}
