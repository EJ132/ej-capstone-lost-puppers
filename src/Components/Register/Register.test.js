import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Register from './Register'

describe(`Register component`, () => {

  it('renders a Register page by default', () => {
    const wrapper = shallow(<Register />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a section with the Register', () => {
    const section = shallow(<Register />)
      .find('.loginBG')
    expect(toJson(section)).toMatchSnapshot()
  })
})
