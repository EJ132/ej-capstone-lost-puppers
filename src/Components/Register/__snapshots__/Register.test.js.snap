// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register component renders a Register page by default 1`] = `
<div>
  <Navbar />
  <section
    className="loginBG"
  >
    <form
      className="login"
      onSubmit={[Function]}
    >
      <section
        className="regError"
        role="alert"
      />
      <img
        alt="paw"
        src="paw.png"
      />
      <label>
        Name
      </label>
      <input
        name="fullname"
        placeholder="EJ Gonzalez"
        required={true}
        type="text"
      />
      <label>
        Username
      </label>
      <input
        name="user_name"
        placeholder="puppy_lover132"
        required={true}
        type="user_name"
      />
      <label>
        Password
      </label>
      <input
        name="password"
        placeholder="********"
        required={true}
        type="password"
      />
      <label>
        Re-enter Password
      </label>
      <input
        name="re_enter_password"
        placeholder="********"
        required={true}
        type="password"
      />
      <button
        type="submit"
      >
        Submit
      </button>
    </form>
    <p
      className="validation"
    >
      <Component />
    </p>
  </section>
  <footer>
    Â© EJ Gonzalez 2019
  </footer>
</div>
`;

exports[`Register component renders a section with the Register 1`] = `
<section
  className="loginBG"
>
  <form
    className="login"
    onSubmit={[Function]}
  >
    <section
      className="regError"
      role="alert"
    />
    <img
      alt="paw"
      src="paw.png"
    />
    <label>
      Name
    </label>
    <input
      name="fullname"
      placeholder="EJ Gonzalez"
      required={true}
      type="text"
    />
    <label>
      Username
    </label>
    <input
      name="user_name"
      placeholder="puppy_lover132"
      required={true}
      type="user_name"
    />
    <label>
      Password
    </label>
    <input
      name="password"
      placeholder="********"
      required={true}
      type="password"
    />
    <label>
      Re-enter Password
    </label>
    <input
      name="re_enter_password"
      placeholder="********"
      required={true}
      type="password"
    />
    <button
      type="submit"
    >
      Submit
    </button>
  </form>
  <p
    className="validation"
  >
    <Component />
  </p>
</section>
`;
